### PR TITLE
Authentication configuration script editor validations

### PR DESCRIPTION
**Pull request by thusithak**
### Proposed changes in this pull request
- Fixes https://github.com/wso2/product-is/issues/3223
screen shot 2018-06-15 at 17 55 20

[Update  5edb534]

screen shot 2018-06-18 at 18 21 52

- Added a submit confirmation message when "update" is clicked with script errors, warnings are ignored.
screen shot 2018-06-15 at 17 55 45

### When should this PR be merged

ASAP

### Follow up actions

None

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [x] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

**Comment by pulasthi7** [comment]
For errors, we should not have an option to proceed.

**Comment by thusithak** [comment]
As per discussion with @pulasthi7 and @ruwanta , 
We decided to disable script evaluation when updating a script with Warnings or Errors.
The erroneous script will still be saved on the editor UI.

screen shot 2018-06-19 at 11 43 36

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-authentication-flow.jsp [hunk 1]
```diff
@@ -670,6 +719,62 @@ var conditionalAuthFunctions = $.parseJSON('<%=availableJsFunctionsJson%>');
     }
 
 
+    function showPopupDialog(message, title, windowHight, okButton, callback, windowWidth) {
```
Comment (pulasthi7) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-authentication-flow.jsp]: typo in height

Also is this different from what we have now with carbonUI?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-authentication-flow.jsp [hunk 2]
```diff
@@ -858,12 +963,42 @@ var conditionalAuthFunctions = $.parseJSON('<%=availableJsFunctionsJson%>');
...
                            onclick="javascript:location.href='configure-service-provider.jsp?display=auth_config&spName=<%=Encode.forUriComponent(spName)%>'"/>
                 </div>
             </form>
         </div>
 
+<div style="display: none;">
+	<div id='messagebox-warning' class="messagebox-warning-custom" style="height: auto;">
+		<h3>Prerequisits</h3>
+		<p>Please change the parameters and default steps as required. For more information, refer
+			<a href="https://docs.wso2.com/display/IS560/Conditional+Authentication">Conditional Authentication Docs</a>
```
Comment (pulasthi7) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-authentication-flow.jsp]: should be 5.7.0 space

Comment (senthalan) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-authentication-flow.jsp]: we change the name to adaptive authentication?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-authentication-flow.jsp (hunk 1 above)
Comment (thusithak) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-authentication-flow.jsp]: Fixed with e5b7774 and b71d7b6

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-authentication-flow.jsp (hunk 2 above)
Comment (thusithak) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-authentication-flow.jsp]: @pulasthi7 fixed with d2a0a20
@senthalan This is a template text that we are soon about to populate with data that are from the script template JSON. So lets keep it as it is for the time being.

**Pull request merged**